nomad/resource_variable.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package nomad
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceVariable() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVariableWrite,
+		Update: resourceVariableWrite,
+		Delete: resourceVariableDelete,
+		Read:   resourceVariableRead,
+		Exists: resourceVariableExists,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"path": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"namespace": {
+				Description: "Variable namespace filter",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     api.DefaultNamespace,
+			},
+			"items": {
+				Type:      schema.TypeMap,
+				Required:  true,
+				Sensitive: true,
+			},
+		},
+	}
+}
+
+func resourceVariableWrite(d *schema.ResourceData, meta any) error {
+	client := meta.(ProviderConfig).client
+
+	variable := api.Variable{
```

Comment:
```suggestion
	variable := &api.Variable{
```

Then we don't need to derefence it below.

Reply:
Forgive me if I misunderstand: As this is not currently a pointer, there is currently no "dereferencing" occurring below. Correct?

May I ask what the advantage of defining variable as a pointer here on line 49 is compared to what the original author did of passing it through as a pointer on line 59?